extensions/2.0/Vendor/MESHOPT_compression/README.md

```diff
@@ -0,0 +1,132 @@
+
+When stored in a GLB file, the dummy buffer should have index 1 or above, to avoid conflicts with GLB binary buffer.
+
+This extension allows buffers to be tagged as fallback by using the `fallback` attribute as follows:
+
+```json
+{
+	"byteLength": 1432878,
+	"extensions": {
+		"MESHOPT_compression": {
+			"fallback": true
+		}
+	}
+}
+```
+
+This is useful to avoid confusion, and can also be used by loaders that support the extension to skip loading of these buffers.
+
+When a buffer is marked as a fallback buffer, the following must hold:
+
+- All references to the buffer must come from `bufferView`s that have a `MESHOPT_compression` extension specified
+- No references to the buffer may come from `MESHOPT_compression` extension JSON
+
+If a fallback buffer doesn't have a URI and doesn't refer to the GLB binary chunk, it follows that `MESHOPT_compression` must be a required extension.
+
+## Compressing geometry data
+
+The codecs used by this extension can represent geometry exactly, replicating both vertex and index data without changes in contents or order. However, to get optimal compression, it's necessary to pre-process the data.
+
+To get optimal compression, encoders should optimize vertex and index data for locality of reference. Specifically:
+
+- Triangle order should be optimized to maximize the recency of previously encountered vertices; this is similar to optimizing meshes for vertex reuse aka post-transform cache in GPU hardware.
+- Vertex order should be linearized in the order that vertices appear in the index stream to get optimal index compression
+
+When index data is not available (e.g. point data sets) or represents topology with a lot of seams (e.g. each triangle has unique vertex indices because it specifies flat-shaded normal), encoders should additionally optimize vertex data for spatial locality, so that vertices close together in the vertex stream are close together in space.
+
+Vertex data should be quantized using the appropriate representation; this extension cleanly interacts with KHR\_quantized\_geometry by compressing already quantized data.
+
+Morph targets can be treated identically to other vertex attributes, as long as vertex order optimization is performed on all target streams at the same time. It is recommended to use quantized storage for morph target deltas, possibly with a narrower type than that used for baseline values.
+
+When storing vertex data, mode 0 (attributes) should be used; for index data, mode 1 (indices) should be used instead. Mode 1 only supports triangle list storage; indices of other topology types can be stored uncompressed or using mode 0 (attributes). The use of triangle strip topology is not recommended since it's more efficient to store triangle lists using mode 1.
+
+TODO: normal & tangent unit filters
+
+## Compressing animation data
+
+To minimize the size of animation data, it is recommended that encoders resample animations before compression. This serves two purposes:

```

Comment:
There are some nuances to consider in this recommendation, such as a file containing multiple animations with different durations. Presumably resampling would happen on a per-animation basis, then? Or share input accessors across all channels of the same duration? I don't know that this needs to be fully specified, but just mention that resampling is not a one-size-fits-all recommendation.

Reply:
Yeah that's fair - it's probably worthwhile to outline several approaches. gltfpack currently resamples each animation in isolation, although it's possible to actually go further and share time tracks between all animations. Regardless, the compression efficiency can probably be gained in other ways - I'll expand/reword this section.